Signup: show a 30 day money back guarantee on the Plans step

mattwiebe opened the pull request:
Adds a new AB test to evaluate the performance of calling out our 30 day money back guarantee on the plans step during signup.

AB test name: `showMoneyBackGuarantee`
Allocations: `50/50`

e2e test PR forthcoming

## To Test

Run `localStorage.setItem( 'ABTests', '{"showMoneyBackGuarantee_20180409":"yes"}' );` in your console before entering signup.

## In Test

<img width="1118" alt="screen shot 2018-04-10 at 10 38 53" src="https://user-images.githubusercontent.com/195089/38573652-5ffab666-3cbc-11e8-8e11-c5fc75b4d27f.png">
<img width="320" alt="screen shot 2018-04-10 at 10 39 42" src="https://user-images.githubusercontent.com/195089/38573653-6020cbc6-3cbc-11e8-8a49-fb7b79ed3582.png">

## Default

<img width="1121" alt="screen shot 2018-04-10 at 12 52 23" src="https://user-images.githubusercontent.com/195089/38574342-3958177c-3cbe-11e8-826a-b898333916d5.png">
<img width="323" alt="screen shot 2018-04-10 at 12 52 43" src="https://user-images.githubusercontent.com/195089/38574343-39793d9e-3cbe-11e8-90bb-8271bd447b6f.png">

fditrapani commented:
Thanks for pinging me on this @mattwiebe. Here are some comments:

- As you mentioned on your post, the spacing does look off. The guarantee followed by the secure payments and then the asterisk-ed message feel like they're also cluttering things up under the plans. Have you thought about grouping the guarantee along with the asterisk so that they're separated by a line break vs a full paragraph space? I also think it might flow better by showing the payment options first and then follow it by the payment guarantee. 
- The green check mark is really clashing with all the payment icons. I wonder if it would work without it, or if a more neutral colour were picked.
- Is it necessary to show the payment options here? What if we replace it with the 30 day guarantee instead? Do you think that would impact how this test is being run? We do show some secure payment messaging on the checkout screen.

akirk commented on client/blocks/plan-footer/index.jsx:
```diff
@@ -0,0 +1,56 @@
+			<div className="plan-footer">
+				<div className="plan-footer__guarantee">
+					<Gridicon icon="checkmark-circle" size="18" />
+					{ translate( '30 day money back guarantee*', {
```
I think it will be easier if we add the `*` outside of the translation.

akirk commented on client/blocks/plan-footer/index.jsx:
```diff
@@ -0,0 +1,56 @@
+					} ) }
+				</div>
+				<div className="plan-footer__fineprint">
+					{ translate( '* Domains purchased with a plan are only refundable for 48 hours.', {
```
Also here we can add the `*` outside of the translation.

mattwiebe commented:
@fditrapani Great feedback above, I've removed the payment icons for the test - it really was a bit too NASCAR down there with everything. Also shrunk the checkmark and made it gray. I fell prey to the "please notice me this is the thing I changed" kind of thinking there. Thoughts on this? 👇

<img width="1120" alt="screen shot 2018-04-11 at 14 51 33" src="https://user-images.githubusercontent.com/195089/38641987-436c00a8-3d9e-11e8-9f29-2840f51227a4.png">
<img width="321" alt="screen shot 2018-04-11 at 14 52 04" src="https://user-images.githubusercontent.com/195089/38641988-439c1838-3d9e-11e8-9abc-fcd2039ee961.png">

fditrapani commented:
Thanks for the updates @mattwiebe. I think it looks much better too. May I suggest that we change the colour of the icon and text so that it doesn't look so out of place. I recommend going with what we're calling [Gray Darken 20](https://wordpress.com/design-handbook/colors/), Here's an example of what it looks like:

<img width="1054" alt="screen shot 2018-04-12 at 9 24 09 am" src="https://user-images.githubusercontent.com/6981253/38680024-47ce80cc-3e33-11e8-9ba7-8ad0a718b9db.png">
 
This might be for another test, or if you're feeling ambitious for this one, but I thought it would be cool to explore something with a little more presence. This is what I came up with:

<img width="1055" alt="screen shot 2018-04-12 at 9 24 57 am" src="https://user-images.githubusercontent.com/6981253/38680105-7c77899a-3e33-11e8-8df9-4ee3d7fa3192.png">

mattwiebe commented:
Great input here @fditrapani - got the banner version in, much nicer personality


![](https://user-images.githubusercontent.com/6981253/38680105-7c77899a-3e33-11e8-8df9-4ee3d7fa3192.png)
<img width="322" alt="screen shot 2018-04-12 at 15 41 07" src="https://user-images.githubusercontent.com/195089/38703275-74e312d2-3e68-11e8-8a11-828083d448a6.png">

fditrapani commented on client/blocks/plan-footer/index.jsx:
```diff
@@ -0,0 +1,64 @@
+	}
+
+	render() {
+		if ( ! this.props.isInSignup || ! this.state.showMoneyBackGuarantee ) {
```
I screwed up my personal hosting so I can't test this on Jetpack connect, do you know if this will show to Jetpack users as well?

mattwiebe replied:
Ah, good question, I will find out.

fditrapani commented on client/blocks/plan-footer/index.jsx:
```diff
@@ -0,0 +1,64 @@
+		return (
+			<div className="plan-footer">
+				<div className="plan-footer__guarantee">
+					{ this.banner( 'left' ) }
```
Semantically this feels little icky to me. It serves no other purpose than visual decoration. I personally would have done it with two pseudo classes in CSS. How would this look/sound for a screen reader?

mattwiebe replied:
Yeah I debated it and my line of thought initially was that I didn't want to weigh down the CSS with the encoded SVG, rather just having it in the JS chunk, but it's pretty lightweight and you've got me swinging the other way.

fditrapani commented:
That looks great @mattwiebe! Thanks for the extra effort to take it to the next level. Looks great on both desktop and mobile. I left a couple comments in the code but otherwise looks amazing!

anneforbush commented:
@mattwiebe Woohoo! Looking good! I may be too late to suggest something (feel free to tell me I am), but wondering if we could change the wording in the banner to make it clear what the guarantee is actually for, as follows:

30-Day Money Back Guarantee On All Plans

mattwiebe commented:
@anneforbush I tested that initially but it worked poorly on small screens. I'll fiddle a bit more to see if I can make it fit

anneforbush commented:
@mattwiebe Ahh gotcha. Thanks for looking into it!
